src/interceptors/exception.interceptor.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  ArgumentsHost,
+  Catch,
+  ExceptionFilter,
+  HttpException,
+  Logger,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { APIResponse } from 'src/utils/apiResponse.utils';
+
+@Catch(HttpException)
+export class HttpExceptionFilter implements ExceptionFilter {
+  private logger = new Logger();
+
+  catch(exception: HttpException, host: ArgumentsHost) {
+    let err = exception['response'];
+    err = err.response ? err.response : JSON.stringify(err);
```

Comment:
PullProofAI

The error message extraction logic can be simplified for better clarity.
```diff
-     let err = exception['response'];
-     err = err.response ? err.response : JSON.stringify(err);
+     const err = exception.getResponse() instanceof Error ? exception.getResponse().message : exception.getResponse();
```


<!-- This is an auto-generated comment by PullProofAI -->